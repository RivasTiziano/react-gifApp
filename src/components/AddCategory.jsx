import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [addCategory, setAddCategory] = useState(""); 

    const handleChange = ({target}) => {
        setAddCategory(target.value.trim());
    };

    const handlePress = ({key, target}) => {
        if(key === 'Enter' && addCategory.length >= 1){
            onNewCategory(addCategory);
            setAddCategory("");
            target.value = "";
        };
    };

    return (
        <input 
            type="text"
            placeholder="Introduce your Category"
            onChange={ handleChange }
            onKeyDown={ handlePress }
        />
  );
};

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}


