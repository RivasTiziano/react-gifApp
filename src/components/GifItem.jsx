import PropTypes from 'prop-types';

export const GifItem = ({title , url}) => {
  return (
    <div className='card'>
        <img data-testid="gif-img" src={url} alt={title}></img>
        <p data-testid="gif-title" >{title}</p>
    </div> 
    )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};
