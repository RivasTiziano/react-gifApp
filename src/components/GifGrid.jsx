import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { PropTypes } from "prop-types";

export const GifGrid = ( {category} ) => {

  const {gifs, loading} = useFetchGifs(category);

  return (
    <>
        {loading && ( <h3>loading...</h3> )}
        <h3>{ category }</h3>
        <div className="card-grid">
          { gifs.map((gifs)=><GifItem key={gifs.id} {...gifs}/>) }
        </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}