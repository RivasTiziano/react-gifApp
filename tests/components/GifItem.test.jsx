import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('Tests in GifItem', () => { 
    
    const gif = {
        id: 1,
        title: "Bob esponja",
        url: "https://media.revistagq.com/photos/5ca5f6a77a3aec0df5496c59/master/w_1600%2Cc_limit/bob_esponja_9564.png"
    }

    test('The snaphot should be pass and throw an error of Proptypes', () => { 

        const {container} = render(<GifItem  {...gif}/>)
        
        expect(container).toMatchSnapshot()

    })

    test('The alt and the src of the img must be the title and the url', () => { 

        render(<GifItem {...gif}/>)

        const {alt, src} = screen.getByTestId('gif-img')

        expect(alt).toBe(gif.title)
        expect(src).toBe(gif.url)

    })

    test('The element must contain the title', () => { 

        render(<GifItem title="Bob esponja"/>)

        const { innerHTML } = screen.getByTestId('gif-title')
           
        expect( innerHTML ).toBe(gif.title)

    })

})