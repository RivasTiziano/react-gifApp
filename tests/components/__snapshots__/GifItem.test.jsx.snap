// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifItem The snaphot should be pass and throw an error of Proptypes 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Bob esponja"
      data-testid="gif-img"
      src="https://media.revistagq.com/photos/5ca5f6a77a3aec0df5496c59/master/w_1600%2Cc_limit/bob_esponja_9564.png"
    />
    <p
      data-testid="gif-title"
    >
      Bob esponja
    </p>
  </div>
</div>
`;
